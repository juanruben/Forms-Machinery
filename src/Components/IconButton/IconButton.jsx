import React from 'react';
import PropTypes from 'prop-types';
import './IconButton.scss';

function IconButton(props) {
    const { icon, onClick } = props;
    return (
        <button onClick={onClick} type="button" className="icon-button">
            <i className={icon} />
        </button>
    );
}

IconButton.propTypes = {
    icon: PropTypes.string.isRequired,
    onClick: PropTypes.func.isRequired,
};

export default IconButton;
