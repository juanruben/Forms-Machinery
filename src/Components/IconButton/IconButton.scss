@import "../../styles.scss";

.icon-button {
  background-color: $icafal-background;
  border: none;
  font-size: 30px;
  padding: 0;

  &:focus {
    outline: 0;
  }

  &:hover {
    color: $icafal-orange-dark;
  }
}
