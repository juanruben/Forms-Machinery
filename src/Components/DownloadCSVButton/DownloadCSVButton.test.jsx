import React from 'react';
import { shallow } from 'enzyme';
import IconButton from './IconButton';

describe('IconButton', () => {
    let wrapper;
    const onClick = jest.fn();

    it('should render ok', () => {
        wrapper = shallow(<IconButton onClick={onClick} icon="" />);
        expect(wrapper.debug()).toMatchSnapshot();
    });

    it('should call callback', () => {
        wrapper = shallow(<IconButton onClick={onClick} icon="" />);
        wrapper.simulate('click');
        expect(onClick).toBeCalledTimes(1);
    });
});
