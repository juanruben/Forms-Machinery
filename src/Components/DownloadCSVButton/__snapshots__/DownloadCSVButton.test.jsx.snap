// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DownloadCSVButton should render ok 1`] = `
"<CSVLink data={{...}} filename=\\"test\\" separator=\\";\\" className=\\"icon-button\\" uFEFF={true} asyncOnClick={false} target=\\"_blank\\">
  <i className=\\"fas fa-file-download\\" />
</CSVLink>"
`;
